plugin-server/src/main/ingestion-queues/session-recordings-consumer.ts

```diff
@@ -306,3 +306,94 @@ const consumedMessageSizeBytes = new Histogram({
     labelNames: ['topic', 'groupId', 'messageType'],
     buckets: exponentialBuckets(1, 8, 4).map((bucket) => bucket * 1024),
 })
+
+// Kafka production related functions using node-rdkafka.
+// TODO: when we roll out the rdkafka library to other workloads, we should
+// likely reuse these functions, and in which case we should move them to a
+// separate file.s
+
+const createKafkaProducer = async (kafkaConfig: KafkaConfig) => {
+    const producer = new RdKafkaProducer({
+        'client.id': hostname(),
+        'metadata.broker.list': kafkaConfig.KAFKA_HOSTS,
+        'security.protocol': kafkaConfig.KAFKA_SECURITY_PROTOCOL
+            ? (kafkaConfig.KAFKA_SECURITY_PROTOCOL.toLowerCase() as Lowercase<KafkaSecurityProtocol>)
+            : 'plaintext',
+        'sasl.mechanisms': kafkaConfig.KAFKA_SASL_MECHANISM,
+        'sasl.username': kafkaConfig.KAFKA_SASL_USER,
+        'sasl.password': kafkaConfig.KAFKA_SASL_PASSWORD,
+        'ssl.ca.pem': kafkaConfig.KAFKA_TRUSTED_CERT_B64
+            ? Buffer.from(kafkaConfig.KAFKA_TRUSTED_CERT_B64, 'base64').toString()
+            : undefined,
+        'ssl.key.pem': kafkaConfig.KAFKA_CLIENT_CERT_B64
+            ? Buffer.from(kafkaConfig.KAFKA_CLIENT_CERT_B64, 'base64').toString()
+            : undefined,
+        'ssl.certificate.pem': kafkaConfig.KAFKA_CLIENT_CERT_KEY_B64
+            ? Buffer.from(kafkaConfig.KAFKA_CLIENT_CERT_KEY_B64, 'base64').toString()
+            : undefined,
+        // milliseconds to wait before sending a batch. The default is 0, which
+        // means that messages are sent as soon as possible. This does not mean
+        // that there will only be one message per batch, as the producer will
+        // attempt to fill batches up to the batch size while the number of
+        // Kafka inflight requests is saturated, by default 5 inflight requests.
+        'linger.ms': 20,
+        // The default is 16kb. 1024kb also seems quite small for our use case
+        // but at least larger than the default.
+        'batch.size': 1024 * 1024, // bytes. The default
+        'compression.codec': 'snappy',
+        dr_cb: true,
+    })
+
+    producer.on('event.log', function (log) {
+        status.debug('📝', 'librdkafka log', { log: log })
+    })
+
+    await new Promise((resolve, reject) =>
+        producer.connect(undefined, (error, data) => {
+            status.info('🔌', 'Connected to Kafka', { error: error, brokers: data.brokers })
```

Comment:
`data` can be undefined on error:
```suggestion
            status.info('🔌', 'Connected to Kafka', { error: error, brokers: data?.brokers })
```